### ISSUE TITLE
Using 'new Datasource()' leads to 'redefine non-configurable property'-error

### ISSUE THREAD

**Issue by queueseven** [dhilt/ngx-ui-scroll#151]
When I use

```
datasource = new Datasource({
    get: ... ,
    settings: { ... }
  });
```

I get the following error at runtime:

> "can't redefine non-configurable property "version""

![grafik](https://user-images.githubusercontent.com/3126597/75546356-a31d0080-5a28-11ea-970c-da63ca05d2da.png)

since version 1.4.0. Version 1.3.4 works fine.

The error is raised here:

https://github.com/dhilt/ngx-ui-scroll/blob/master/src/component/classes/adapter.ts#L93

**Comment by dhilt** [comment]
@queueseven Thanks for the report! I released ngx-ui-scroll v1.4.2 containing necessary fix for the issue. Would be great if you confirm it works for you.

**Comment by queueseven** [comment]
@dhilt Seems to work now. Thanks for the quick fix!

**Comment by queueseven** [comment]
Sorry, that was to fast :-)

It seemed to work at first, but now I see

> ERROR TypeError: "can't redefine non-configurable property "_isLoading$""

**Comment by dhilt** [comment]
@queueseven hm...looks like observable props are also affected

**Comment by dhilt** [comment]
@queueseven Can't catch the `$isLoading` case, in my vision it should be safe enough... the dashed observable props are being set only once. Could you please reproduce your infrastructure forking [this v1.4.2 stackblitz](https://stackblitz.com/edit/angular-ngx-ui-scroll-1-4-2)? dependencies and the lib usage. It would be very helpful!

**Comment by queueseven** [comment]
Sorry, I can't repoduce it myself, since I changed the code I'm currently working on a lot. I think one of the problem was my complicated template/component with multiple `*uiScroll` and trying to change the `get` function of my datasource/datasources at runtime. Now I simply use a single `*uiScroll` that I never hide and everything works fine.

**Comment by dhilt** [comment]
@queueseven That's it... By the way, I arranged a new issue recently that might relate to what you said on changing the `datasource` at runtime: [Replace Datasource on the fly](https://github.com/dhilt/ngx-ui-scroll/issues/139).
